mongo/datadog_checks/mongo/api.py

```diff
@@ -111,3 +115,23 @@ def _get_deployment_type(self):
             return ReplicaSetDeployment(replset_name, replset_state, cluster_role=cluster_role)
 
         return StandaloneDeployment()
+
+    def _get_alibaba_deployment_type(self):
+        is_master_payload = self['admin'].command('isMaster')
+        if is_master_payload.get('msg') == 'isdbgrid':
+            return MongosDeployment()
+        # On alibaba cloud, a mongo node is either a mongos or part of a replica set.
+        repl_set_payload = self['admin'].command("replSetGetStatus")
+        replset_name = repl_set_payload["set"]
+        replset_state = repl_set_payload["myState"]
```

Comment:
nit: worth abstracting since this logic is also repeated in `_get_deployment_type`?